File: server/src/gql/graphql.py
"""
GraphQL related python types, other than MongoDB models.

Keep in sync with schema.graphql (also with graphql.ts)
"""
from typing import TypedDict


class NewMatesInput(TypedDict):
    """New Mate Input class passed from client."""

    name: str
    lastSynced: str  # date gets converted to str


class MateAssignmentInput(TypedDict):
    """GQL Input for assigning mates to a different family."""

    mateId: str
    fromFamilyId: str
    toFamilyId: str

File: client/src/graphql.ts
// TS version of schema.graphql (also keep in sync with graphql.py)
import { gql } from "@apollo/client";

/**
 * GQL for adding new mates
 */
export const ADD_NEW_MATES = gql`
  mutation addMatesFn($newMates: [NewMatesInput]!) {
    addNewMates(mates: $newMates) {
      id
      name
    }
  }
`;

/**
 * Return type of addNewMates
 */
export type AddNewMatesReturn = Pick<Mate, "id" | "name">;

/**
 * Input type for addNewMates
 */
export type NewMatesInput = {
  name: string;
  lastSynced: Date;
};

/**
 * GQL Typescript type for Mate
 */
export type Mate = {
  id: string;
  name: string;
  sync_ids: [string];
};
